src/Html/Events/Extra.elm

```diff
@@ -235,34 +222,25 @@ targetSelectedIndex =
 -}
 onClickPreventDefault : msg -> Attribute msg
 onClickPreventDefault msg =
-    onWithOptions "click"
-        { defaultOptions | preventDefault = True }
-        (Json.succeed msg)
+    custom "click" (Json.succeed { message = msg, stopPropagation = False, preventDefault = True })
 
 
 {-| Always send `msg` upon click, preventing click propagation.
 -}
 onClickStopPropagation : msg -> Attribute msg
 onClickStopPropagation msg =
-    onWithOptions "click"
-        { defaultOptions | stopPropagation = True }
-        (Json.succeed msg)
+    custom "click" (Json.succeed { message = msg, stopPropagation = True, preventDefault = False })
```

Comment:
Related to the previous comment, I think we should probably be using `Html.Events.stopProagationOn` here instead of `custom`.